src/field-value.js

```diff
@@ -92,9 +80,86 @@ class FieldValue {
   }
 }
 
-DELETE_SENTINEL = new FieldValue();
-SERVER_TIMESTAMP_SENTINEL = new FieldValue();
+/**
+ * An internal interface shared by all field transforms.
+ *
+ * A 'FieldTransform` subclass should implement '.includeInDocumentMask',
+ * '.includeInDocumentTransform' and 'toProto' (if '.includeInDocumentTransform'
+ * is 'true').
+ *
+ * @private
+ * @abstract
+ */
+class FieldTransform extends FieldValue {}
```

Comment:
Should the parent type include `includeInDocumentMask` and `includeInDocumentTransform`?

You seem to be assuming that they exist for all `FieldTransform` instances.

Reply:
I was going to point this out in the PR description, but forgot: With linting enabled and 100 code coverage as a requirement, it is not actually possible to add 'abstract' methods (aka: fake them by adding an unused method that throws an exception). When I move this codebase to TypeScript, this will become an interface, which will solve both of these issues.